src/Widgets/CameraView.vala

```diff
@@ -118,6 +120,8 @@ public class Camera.Widgets.CameraView : Gtk.Stack {
                 camera_removed (device);
                 cameras.remove ((owned) device);
```

Comment:
This does not remove the camera when started with it enabled and then disabled by the keyboard.  However, it works properly if started with camera disabled, then enabled, then disabled again.  I would recommend you use a HashMap and identify the devices according to their display name or other unchangable property rather than relying on the object matching.

Reply:
I did test it with no camera and then plugging in, so I'm glad it worked for you as well. I'll try starting with a camera and then disconnecting too.